### PR TITLE
Adding vm user list test cases

### PR DESCRIPTION
**Pull request by rananda**

### CODE REVIEW

File: cfme/infrastructure/virtual_machines.py [hunk 1]
```diff
@@ -345,6 +353,40 @@ def does_vm_exist_in_cfme(self):
...
+        #find_quadicon(vm)
+        #find_quadicon(vm_name, do_not_navigate=False)
+        cfg_btn('Set Ownership')
+        if click_reset:
+            fill(self.set_ownership_form, {'user_name': user,
+                                           'group_name': group},
+                action=form_buttons.reset)
+            flash.assert_message_match('All changes have been reset')
+
+        if click_cancel:
```
Comment (psav) [inline cfme/infrastructure/virtual_machines.py]: Isn't this supposed to be part of the previous conditional block?

File: cfme/infrastructure/virtual_machines.py [hunk 2]
```diff
@@ -345,6 +353,40 @@ def does_vm_exist_in_cfme(self):
...
+                action=form_buttons.cancel)
+            flash.assert_success_message('Set Ownership was cancelled by the user')
+        else:
+            fill(self.set_ownership_form, {'user_name': user,
+                                           'group_name': group},
+                action=form_buttons.add)
+            flash.assert_success_message('Ownership saved for selected Virtual Machine')
+
+    def unset_ownership(self, vm=None):
+        #choose the vm code comes here
```
Comment (psav) [inline cfme/infrastructure/virtual_machines.py]: Assuming this is not done yet....

**Comment by dajoRH** [comment]
Lint report for commit 5aba493800b71976622af67ccaec077b59855920:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/configure/test_user_vm_list.py:68:5: E265 block comment should start with '# '
./cfme/tests/configure/test_user_vm_list.py:69:5: E265 block comment should start with '# '
./cfme/infrastructure/virtual_machines.py:357:9: E265 block comment should start with '# '
./cfme/infrastructure/virtual_machines.py:358:9: E265 block comment should start with '# '
./cfme/infrastructure/virtual_machines.py:359:9: E265 block comment should start with '# '
./cfme/infrastructure/virtual_machines.py:360:9: E265 block comment should start with '# '
./cfme/infrastructure/virtual_machines.py:361:9: E265 block comment should start with '# '
./cfme/infrastructure/virtual_machines.py:363:9: E265 block comment should start with '# '
./cfme/infrastructure/virtual_machines.py:364:9: E265 block comment should start with '# '
./cfme/infrastructure/virtual_machines.py:384:9: E265 block comment should start with '# '
```
*CFME QE Bot*

File: cfme/infrastructure/virtual_machines.py (hunk 2 above)
Comment (rananda) [inline cfme/infrastructure/virtual_machines.py]: yes it is yet to be coded

File: cfme/tests/configure/test_user_vm_list.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                email='abc@redhat.com',
+                group=new_group,
+                cost_center='Workload',
+                value_assign='Database')
+    user.create()
+    return user
+
+
+@pytest.fixture(scope="module")
+def test_set_user_ownership_on_vm(request, new_role, new_group):
```
Comment (psav) [inline cfme/tests/configure/test_user_vm_list.py]: If ```new_group``` requires ```new_role```, then you don't need ```new_role``` in this test definition.

**Comment by dajoRH** [comment]
Lint report for commit 6bd230e306db065d2589b509f584a2b2892800c7:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/infrastructure/virtual_machines.py:366:9: E265 block comment should start with '# '
./cfme/infrastructure/virtual_machines.py:366:9: E303 too many blank lines (7)
./cfme/infrastructure/virtual_machines.py:367:9: E265 block comment should start with '# '
./cfme/infrastructure/virtual_machines.py:368:9: E265 block comment should start with '# '
./cfme/infrastructure/virtual_machines.py:372:1: W293 blank line contains whitespace
./cfme/infrastructure/virtual_machines.py:396:9: E265 block comment should start with '# '
./cfme/tests/configure/test_user_vm_list.py:73:1: E302 expected 2 blank lines, found 1
./cfme/tests/configure/test_user_vm_list.py:76:56: W292 no newline at end of file
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please?
*CFME QE Bot*

File: cfme/tests/configure/test_user_vm_list.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import cfme.configure.access_control as ac
```
Comment (dajohnso) [inline cfme/tests/configure/test_user_vm_list.py]: @rananda lets move this under the infrastructure folder and rename to test_vm_ownership.py

**Comment by dajoRH** [comment]
Lint report for commit 861e396c26f7003d650cd90462a34248edbbba98:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/configure/test_user_vm_list.py:68:5: E265 block comment should start with '# '
./cfme/tests/configure/test_user_vm_list.py:69:5: E265 block comment should start with '# '
```
:trollface:
*CFME QE Bot*

File: cfme/infrastructure/virtual_machines.py [hunk 5]
```diff
@@ -492,6 +500,29 @@ def _assign_unassign_policy_profiles(self, assign, *policy_profile_names):
         sel.move_to_element('#tP')
         form_buttons.save()
 
+    def set_ownership(self, user=None, group=None, click_cancel=False, click_reset=False):
+        sel.click(self.find_quadicon(False, False, False))
+        cfg_btn('Set Ownership')
+        if click_reset:
```
Comment (psav) [inline cfme/infrastructure/virtual_machines.py]: OK, so we can do this in a nicer way
```python
     if click_reset:
         action = form_buttons.reset
         msg_assert = lambda: flash.assert_message_match('All changes have been reset')
     elif click_cancel:
         action = form_buttons.cancel
         msg_assert = lambda: flash.assert_success_message('Set Ownership was cancelled by the user')
     else
         action = form_buttons.add
         msg_assert = lambda: flash.assert_success_message('Ownership saved for selected Virtual Machine')

     fill(set_ownership_form, {'user_name': user, 'group_name': group},
             action=action)
     msg_assert()
```

File: cfme/tests/configure/test_user_vm_list.py (hunk 4 above)
Comment (psav) [inline cfme/tests/configure/test_user_vm_list.py]: @rananda this still needs to be done

Comment (psav) [inline cfme/tests/configure/test_user_vm_list.py]: Oh I just saw...I think you just need to remove this file

**Comment by dajoRH** [comment]
Lint report for commit e20abd1706fa31af9edf86d4332ac1c2ca3096f2:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/configure/test_user_vm_list.py:68:5: E265 block comment should start with '# '
./cfme/tests/configure/test_user_vm_list.py:69:5: E265 block comment should start with '# '
```
:trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 68dbbc7269a3268495a5695b047bae22472c9968:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/infrastructure/test_vm_ownership.py:78:5: F821 undefined name 'set_vm_user'
./cfme/tests/configure/test_user_vm_list.py:68:5: E265 block comment should start with '# '
./cfme/tests/configure/test_user_vm_list.py:69:5: E265 block comment should start with '# '
```
:trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c9a02a249d331a426dd92487b4028e8711e10cfc:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/configure/test_user_vm_list.py:68:5: E265 block comment should start with '# '
./cfme/tests/configure/test_user_vm_list.py:69:5: E265 block comment should start with '# '
```
:trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c8ad892ff46239a9c435365968613fea58295012:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/configure/test_user_vm_list.py:68:5: E265 block comment should start with '# '
./cfme/tests/configure/test_user_vm_list.py:69:5: E265 block comment should start with '# '
```
:trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 157e2c0e03e37c6c02f8ec5cc19a8ca6652a6f9b:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/infrastructure/test_vm_ownership.py:7:1: F401 'pdb' imported but unused
./cfme/tests/configure/test_user_vm_list.py:68:5: E265 block comment should start with '# '
./cfme/tests/configure/test_user_vm_list.py:69:5: E265 block comment should start with '# '
```
:trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 220e18eceb8f216b0c956eda57371485fd385b6c:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 384f9b1ab2aaafc276af71060d716dca20d14918:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c52f86bb856811d8b60ffb57552b2c443a381848:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 2edb09f5acfb8cda2d302bd9d69457a8fc91fd9f:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d79f9ba5676db3131ba0b86e31cb208818cad475:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**